pylint/interfaces.py

```diff
@@ -79,6 +79,18 @@ class IAstroidChecker(IChecker):
     """
 
 
+class IAstroidAndTokensChecker(IAstroidChecker):
```

Comment:
Isn't `BaseTokenChecker` exact same thing? From what I see `BaseTokenChecker` _is_ a checker, which means it can involve walking ASTs and visiting nodes... Or am I missing something?

Reply:
Yes, well spotted

Reply:
There is a side effect however:
making `StringFormatChecker` inherits from `BaseTokenChecker` and implementing `ITokenChecker` forces to change all the `.add_message` call sites in this class, because it now requires a `line` parameter:
https://github.com/PyCQA/pylint/blob/master/pylint/utils.py#L374
https://github.com/PyCQA/pylint/blob/master/pylint/utils.py#L153